pkg/operator/provisioner/provisioner.go

```diff
@@ -141,6 +141,12 @@ func (p *RookVolumeProvisioner) createVolume(image, pool string, size int64) err
 // by the given PV.
 func (p *RookVolumeProvisioner) Delete(volume *v1.PersistentVolume) error {
 	logger.Infof("Deleting volume %s", volume.Name)
+	if volume.Spec.PersistentVolumeSource.FlexVolume == nil {
+		return fmt.Errorf("Failed to delete rook block image %s/%s: %v", p.provConfig.pool, volume.Name, "PersistentVolume is not a FlexVolume")
```

Comment:
this is probably wise to add these checks @Ulexus, thanks!  out of curiosity regarding the style of your errors, is there a clear advantage to adding the last portion of the error message as a `%v` format arg instead of just putting the string inline a la:
```
return fmt.Errorf("Failed to delete rook block image %s/%s: PersistentVolume is not a FlexVolume", p.provConfig.pool, volume.Name)
```
i hadn't come across the style you are using, so I'm curious about the advantage of it.  Thanks!

Reply:
Not at all; the only reason I did that is so that the bulk of the text of the returned Error would be the same as the previous... just in case something else does a text search (I really hope that is not the case).

Reply:
I would generally not put the summary of the function in the error text returned by that function.  I would only return the "interesting" part of the error.  One presumes the caller of the function would know what the function was to do, and if an error was returned, that means that the function failed to do that.  (This is not universally the case, but it is _generally_ expected.)  Thus here, I would just return the important part of the failure... "PersistentVolume is not a FlexVolume" or whatever.  The calling function would wrap any received errors in the appropriate context ("failed to delete block image").  

In my own code, I make copious use of [`errors.Wrap()`](https://godoc.org/github.com/pkg/errors#Wrap) for cases just like this... but I'm certainly not going to introduce a new dependency and rewrite your standard patterns just for this. ;)